src/NServiceBus.Core/Routing/RoutingComponent.cs

```diff
@@ -48,19 +42,19 @@ public static RoutingComponent Initialize(Configuration configuration,
         hostingConfiguration.Services.AddSingleton(sp =>
             new UnicastSendRouter(
                 isSendOnlyEndpoint,
-                receiveConfiguration.QueueNameBase,
+                receiveConfiguration.LocalQueueAddress.BaseAddress,
                 receiveConfiguration.InstanceSpecificQueueAddress,
                 distributionPolicy,
                 unicastRoutingTable,
                 endpointInstances,
-                transportSeam.TransportAddressResolverBuilder(sp)));
+                sp.GetRequiredService<ITransportAddressResolver>()));
```

Comment:
I know we had some idea to make the reference to transportseam explicit but I don't see enough ROI?